src/cli/list.rs

```diff
@@ -0,0 +1,195 @@
+            .map(|p| p.as_source().to_string())
+            .collect()
+    };
+
+    // Convert the the list of package record to specific output format
+    let mut packages_to_output = locked_deps
+        .iter()
+        .map(|p| {
+            match p.kind {
+                // Conda package
+                LockedDependencyKind::Conda(_) => {
+                    let name = p.name.clone();
+                    let version = p.version.clone();
+                    let kind = "conda".to_string();
+                    let build = p.as_conda().unwrap().build.clone();
+                    // NOTE(hadim): channel is not available, only the source url...
+                    // let channel = Some(p.as_conda().unwrap().url.to_string().clone());
+                    let channel = Some("".to_string());
+                    let is_explicit = project_dependency_names.contains(&name);
+
+                    PackageToOutput {
+                        name,
+                        version,
+                        build,
+                        kind,
+                        channel,
+                        is_explicit,
+                    }
+                }
+                // Pypi package
+                LockedDependencyKind::Pypi(_) => {
+                    let name = p.name.clone();
+                    let version = p.version.clone();
+                    let kind = "pypi".to_string();
+                    let build = p.as_pypi().unwrap().build.clone();
+                    let channel = Some("".to_string());
+                    let is_explicit = project_dependency_names.contains(&name);
+
+                    PackageToOutput {
+                        name,
+                        version,
+                        build,
+                        kind,
+                        channel,
+                        is_explicit,
+                    }
+                }
+            }
+        })
+        .collect::<Vec<_>>();
+
+    // Filter packages by regex if needed
+    if let Some(regex) = args.regex {
+        let regex = regex::Regex::new(&regex).map_err(|_| miette::miette!("Invalid regex"))?;
+        packages_to_output = packages_to_output
+            .into_iter()
+            .filter(|p| regex.is_match(&p.name))
+            .collect::<Vec<_>>();
+    }
+
+    // Sort packages by name if needed
+    if !args.no_sort {
+        // Sort packages by name
+        packages_to_output.sort_by(|a, b| a.name.cmp(&b.name));
+    }
+
+    // Print as table string or JSON
+    if args.json {
+        // print packages as json
+        json_packages(packages_to_output, args.json_pretty);
+    } else {
+        // print packages as table
+        print_packages(packages_to_output);
+    }
+
+    Ok(())
+}
+
+fn print_packages(packages: Vec<PackageToOutput>) {
+    // NOTE(hadim): maybe switch to a third party package or a custom dynamic printing logic
+    // to adapt the table column width to the content of the table and the terminal.
```

Comment:
The current printed table works nicely, but please let me know what you think about a more dynamic table printing (maybe relying on an external crate).

Reply:
I gave it a try to comfy_table and it works nicely. But it also adds a new dep. Let me know what you think.